### DISCUSSION TITLE
how to enable github copilot free subscription

### DISCUSSION THREAD

**Discussion by ghost** [community/community#31133] [Copilot Conversations] [answered]
Hi, after 60 days of adding github copilot extension to visual studio code IDE, I can't use github copilot.
I know it's free for students and  maintainers of popular open-source repositories.
But I'm not a student.
I want to know how many repos should I create in my github account at least so that I can use the github copilot extension for free?
I really want to use the github copilot extension, so please answer my question

**Comment by melvincwng** [comment, marked as answer]
@Ramil93 Hey Ramil, as you clearly pointed out there's only two ways to use GitHub CoPilot for free - either a) Being a student or b) being a maintainer of popular open source repositories.

If you do not fall into any of those two categories, the only option available left for you would be to pay for GitHub Copilot.

Regarding this part of your post:
> I want to know how many repos should I create in my github account at least so that I can use the github copilot extension for free?

Assuming I understood correctly, unfortunately it doesn't matter how many repos you create... You need to be either a student or a popular open source repository maintainer to use CoPilot for free.

Hope this answers your question.

**Reply by ghost** [reply]
@melvincwng I realized that regardless of the number of repos I uploaded to my Github account, it should be popular.
If I upload a popular repo to my github account, after how long will I be able to use github copilot for free?
Does the system automatically confirm this or github colleagues?
Thanks for your answers.

**Reply by melvincwng** [reply]
@Ramil93 Hey Ramil, unfortunately there does not seem to be any official guidelines or metrics to determine who qualifies as a popular open source repository maintainer.

Based on what I seen, I would not put too much emphasis on this particular option. There are instances where I saw various discussion threads whereby some users who claim they are maintainers of popular open source repositories, and yet they didn't get access to use GitHub Copilot for free.

**Comment by ghost** [comment]
@melvincwng I understood, thank you for informing me about this.

**Reply by melvincwng** [reply]
@Ramil93 No problem, glad to have helped you 😁
